### PR TITLE
Events to micronaut version 4.3.4

### PR DESCRIPTION
**Pull request by ChaimaaeROUAI**

### CODE REVIEW

File: src/events/app/src/main/java/events/model/Event.java [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * The event to track.
+ */
+@Serdeable
+public class Event {
```
Comment (alvarosanchez) [inline src/events/app/src/main/java/events/model/Event.java:13]: Record please

File: src/events/app/src/main/java/events/model/EventRecord.java [hunk 2]
```diff
@@ -1,48 +1,48 @@
...
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Event record.
+ */
+@Serdeable
+public class EventRecord extends Event {
```
Comment (alvarosanchez) [inline src/events/app/src/main/java/events/model/EventRecord.java:14]: Record please

File: src/events/app/src/main/java/events/model/EventsReceived.java [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
-}
+package events.model;
+
+import io.micronaut.serde.annotation.Serdeable;
+
+/**
+ * Defines whether events were received.
+ */
+@Serdeable
+public class EventsReceived {
```
Comment (alvarosanchez) [inline src/events/app/src/main/java/events/model/EventsReceived.java:9]: Record

File: src/events/app/src/main/resources/application-app.yml [hunk 4]
```diff
@@ -1,3 +1,7 @@
-kafka:
-  bootstrap:
-    servers: localhost:9092
+kafka:
+  bootstrap:
+    servers: localhost:9092
+test-resources:
+ containers:
+   kafka:
+     image-name: "confluentinc/cp-kafka:7.3.3"
```
Comment (alvarosanchez) [inline src/events/app/src/main/resources/application-app.yml:7]: Do we need this specific version? Can we not use Test Resources' default?

File: src/events/app/src/main/resources/application-app.yml [hunk 5]
```diff
@@ -1,3 +1,7 @@
-kafka:
-  bootstrap:
-    servers: localhost:9092
+kafka:
+  bootstrap:
+    servers: localhost:9092
```
Comment (alvarosanchez) [inline src/events/app/src/main/resources/application-app.yml:3]: Remove, this is provided by Test Resources

**Review by alvarosanchez — CHANGES_REQUESTED** [review]

File: src/events/app/src/main/java/events/model/EventRecord.java (hunk 2 above)
Comment (ChaimaaeROUAI) [reply src/events/app/src/main/java/events/model/EventRecord.java]: done

File: src/events/app/src/main/java/events/model/EventsReceived.java (hunk 3 above)
Comment (ChaimaaeROUAI) [reply src/events/app/src/main/java/events/model/EventsReceived.java]: done

File: src/events/app/src/main/resources/application-app.yml (hunk 4 above)
Comment (ChaimaaeROUAI) [reply src/events/app/src/main/resources/application-app.yml]: sure we can

**Review by alvarosanchez — APPROVED** [review]

**Review by GH-OIM — APPROVED** [review]

**Review by Julia3072 — APPROVED** [review]

**Pull request merged**